tests/test_homebrew_workflow_security.py

```
@@ -0,0 +1,358 @@
+                env=env,
+                text=True,
+                capture_output=True,
+                check=False,
+            )
+            if output_path.exists():
+                result.stdout += "\n" + output_path.read_text(encoding="utf-8")
+            return result
+
+    def test_workflow_contract_is_secret_and_runner_safe(self) -> None:
+        text = WORKFLOW.read_text(encoding="utf-8")
+        document = yaml.load(text, Loader=yaml.BaseLoader)
```

Ruff (0.16.3)

[error] Probable use of unsafe loader `BaseLoader` with `yaml.load`. Allows instantiation of arbitrary objects. Consider `yaml.safe_load`.

(S506)